### PR TITLE
Add footer navigation

### LINKED ISSUE #117: Add social media icon styling to menu widgets
**Is your feature request related to a problem? Please describe.**
I often want to add social menus to my site. This can be achieved using the beta widget-block screen, but it would also be nice to support social icons inside menu widgets.

**Describe the solution you'd like**
If you link to a social media account within your menu widget, the link is transformed into a social media icon.

**Describe alternatives you've considered**
Only supporting the social icon block, or not including support for social menus — but neither is ideal, IMO.

**Additional context**
Maybe we could reuse the SVGs and styles packaged with the Gutenberg social icon block?

This is a low-priority task.

### LINKED ISSUE #148: Add navigation menu to the footer
Add a navigation menu to the footer of the theme. This could replace the `.site-name` div.

The menu could serve for general links, as well as to add a link to the privacy policy page (perhaps automatically?).

Screenshot 2020-09-25 at 15 14 09

### PR DESCRIPTION
**Pull request by carolinan**
Fixes #148 
Fixes #117 

## Summary

Registers a new menu location called Footer.
Adds the nav menu to the footer between the site info and widget area.
Prepares a nav menu walker for social icons
Adds initial SCSS for the menu.
Adds the social icon svgs.

A screen reader text is used together with the icons.

_TLDR; It adds back the social menu that was in Seedlet,
but makes sure that mixed content can be used._

## Relevant technical choices:
Used the same hover style for the icons as the social icon block.

## Test instructions
This PR can be tested by following these steps:
1.  Create a new menu with mixed content, both a regular link and a social media link.
2. Add the menu to the new footer menu location
3. View the front. Test that icons are showing.

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

## Screenshot:
footer

Icon link with focus:
icon-focus

### CODE REVIEW

**Review by rolf-yoast — APPROVED** [review]
Code review 👍

**Comment by melchoyce** [comment]
Looking good!

Can we reduce the spacing between the footer and the menu, and instead increase the spacing between the menu and the widgets? So it looks like this:

image

Also, this is more of an "is it possible" question, but can we center any remaining items on the last line of the wrapped menu using flex or something?

image

**Review by melchoyce — CHANGES_REQUESTED** [review]
Looking good!

Can we reduce the spacing between the footer and the menu, and instead increase the spacing between the menu and the widgets? So it looks like this:

image

Also, this is more of an "is it possible" question, but can we center any remaining items on the last line of the wrapped menu using flex or something?

image

File: assets/sass/06-components/footer-navigation.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+.footer-navigation {
+	font-size: var(--global--font-size-xs);
+	margin-top: calc(2 * var(--global--spacing-vertical));
+	text-align: center;
```
Comment (carolinan) [inline assets/sass/06-components/footer-navigation.scss]: With text-align: center, here, is 	`justify-content: center;` on the UL .footer-navigation-wrapper still needed?
I am not sure this is correct.

**Comment by carolinan** [comment]
My _after_ screenshot with the new margins and text-align: center:

footer-nav-center-wrap

File: assets/sass/06-components/footer-navigation.scss (hunk 1 above)
Comment (rolf-yoast) [reply assets/sass/06-components/footer-navigation.scss:5]: When I removed the `justify-content: center;` it didn't align in the center anymore. But when I removed the `text-align: center;` and kept the other one it still aligned centered. So the question is if the `text-align: center` is needed.

Comment (rolf-yoast) [reply assets/sass/06-components/footer-navigation.scss]: I made a small commit to remove it.

**Review by rolf-yoast — APPROVED** [review]
Acceptance 👍

**Pull request merged**